Sources/AsyncHTTPClient/ConnectionPool/HTTP1/HTTP1ClientChannelHandler.swift

```diff
@@ -269,8 +269,9 @@ final class HTTP1ClientChannelHandler: ChannelDuplexHandler {
             }
 
             oldRequest.succeedRequest(buffer)
+            promise?.succeed(())
```

Comment:
this promise should be passed to:

```swift
            case .sendRequestEnd:
                context.writeAndFlush(self.wrapOutboundOut(.end(nil)), promise: nil)
```

this will be more obvious once we attach the promise to `sendRequestEnd` and not to `succeedRequest`. In this case we should also defer the sending of `oldRequest.succeedRequest(buffer)` until the promise has fired, since we may run into ordering issues otherwise.